src/kernel/chainstatemanager_opts.h

```diff
@@ -27,6 +27,11 @@ namespace kernel {
 struct ChainstateManagerOpts {
     const CChainParams& chainparams;
     const std::function<NodeClock::time_point()> adjusted_time_callback{nullptr};
+    /**
+     * How many dedicated script-checking threads are running.
+     * 0 indicates all script checking is done on the main threadMessageHandler thread.
```

Comment:
> In the future it would be good to change the kernel to read this option and call StartScriptCheckWorkerThreads directly.

I wonder where this should be done, ideally. Doing it in the `for` loop that constructs the chainstatemanager and calls `LoadChainstate`, ... may cause it being called several times, as it is a loop. Though, we only want to start the threads once.

Reply:
re: https://github.com/bitcoin/bitcoin/pull/25704#discussion_r957183985

> I wonder where this should be done, ideally. Doing it in the `for` loop that constructs the chainstatemanager and calls `LoadChainstate`, ... may cause it being called several times, as it is a loop. Though, we only want to start the threads once.

Maybe a simpler approach is to drop the `script_check_threads` variable and use `scriptcheckqueue.m_worker_threads.size()` in validation code instead.

Reply:
(this was done, so closing thread for now)